From d69e0a4d3ed0955b5b257bcc153f62ff4e8cd837 Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Tue, 23 Jul 2019 11:03:26 +0800
Subject: [PATCH 1146/1214] parse evmm for intel-iommu, for WA the VTD enabled
 by evmm

VTD enabled by evmm, no DMAR device will exposed to Linux
this is used to inform i915 that VTD enabled by evmm, so
WA will be applied

Change-Id: I09e96b1abe0c8a3796562eb2d59ce4239c7b1103
Tracked-On: OAM-83938
Signed-off-by: Yang, Dong <dong.yang@intel.com>
---
 drivers/iommu/intel-iommu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index dd424d0..d8683f8 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -555,6 +555,10 @@ static int __init intel_iommu_setup(char *str)
 			printk(KERN_INFO
 				"Intel-IOMMU: not forcing on after tboot. This could expose security risk for tboot\n");
 			intel_iommu_tboot_noforce = 1;
+		} else if (!strncmp(str, "evmm", 4)) {
+			printk(KERN_INFO
+				"Intel-IOMMU: enabled by evmm\n");
+			intel_iommu_gfx_mapped = 1;
 		}
 
 		str += strcspn(str, ",");
-- 
2.7.4

